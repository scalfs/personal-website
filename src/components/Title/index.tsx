import { HTMLAttributes } from 'react'

const Title: React.FC<HTMLAttributes<HTMLHeadingElement>> = ({
  children,
  ...props
}) => (
  <h1
    {...props}
    className="mb-8 text-4xl font-extrabold tracking-tight sm:text-5xl lg:text-6xl"
  >
    {children}
  </h1>
)

export default Title
