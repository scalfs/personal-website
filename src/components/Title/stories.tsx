import { ComponentMeta, ComponentStory } from '@storybook/react'
import Title from '.'

export default {
  title: 'Title',
  component: Title
} as ComponentMeta<typeof Title>

const Story: ComponentStory<typeof Title> = () => <Title>Heading</Title>

export const Default = Story.bind({})
