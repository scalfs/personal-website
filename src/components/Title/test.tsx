import { render, screen } from '@testing-library/react'

import Title from '.'

describe('<Title />', () => {
  it('should render the title', () => {
    const { container } = render(<Title>Heading</Title>)

    expect(
      screen.getByRole('heading', { name: /Heading/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
