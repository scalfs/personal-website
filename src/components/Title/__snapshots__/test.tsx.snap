// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render the title 1`] = `
<h1
  class="mb-4 text-4xl font-extrabold tracking-tight sm:mb-8 sm:text-5xl lg:text-6xl"
>
  Heading
</h1>
`
